bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDCyclicTransposeFlowGraphFinder.java

```diff
@@ -0,0 +1,233 @@
+            List<String> previousNodesInTransposeFlow) {
+
+        var copyPreviousNodesInTransposeFlow = new ArrayList<>(previousNodesInTransposeFlow);
+
+        List<DFDVertex> result = new ArrayList<>();
+
+        Node previousNode = incomingFlow.getSourceNode();
+
+        List<Pin> previousNodeInputPins = getAllPreviousNodeInputPins(previousNode, incomingFlow);
+
+        List<DFDVertex> previousNodeVertices = new ArrayList<>();
+
+        if (!loopCheck(copyPreviousNodesInTransposeFlow, previousNode.getEntityName())) {
+            if (!hasCycles) {
+                logger.warn("Resolving cycles: Stoping cyclic behavior for analysis, may cause unwanted behavior");
+                hasCycles = true;
+            }
+            copyPreviousNodesInTransposeFlow.add(previousNode.getEntityName());
+            previousNodeVertices = loopAwareDetermineSinks(new DFDVertex(previousNode, new HashMap<>(), new HashMap<>()), previousNodeInputPins, sourceNodes,
+                    copyPreviousNodesInTransposeFlow);
+            
+        } else {
+            copyPreviousNodesInTransposeFlow.add(previousNode.getEntityName());
+            previousNodeVertices = determineSinks(new DFDVertex(previousNode, new HashMap<>(), new HashMap<>()), previousNodeInputPins, sourceNodes,
+                    copyPreviousNodesInTransposeFlow);
+        }
+        
+        for (DFDVertex vertex : finalVertices) {
+            result.addAll(cloneFlowAndVertexForMultipleFlowGraphs(vertex, inputPin, incomingFlow, previousNodeVertices));
+        }
+        return result;
+    }
+
+    /**
+     * checks if the source of incoming flow is part of a loop (We allow first iteration of loop)
+     * @param previousNodesInTransposeFlow List of all Nodes part of the current transpose Flow
+     * @param sourceNode
+     */
+    private boolean loopCheck(List<String> previousNodesInTransposeFlow, String sourceNode) {
+        long count = previousNodesInTransposeFlow.stream()
+                .filter(item -> item.equals(sourceNode))
+                .count();
+        return count < ITERATIONS_OF_LOOP;
+    }
+
+    /**
+     * Clones a vertex with its predecessors to use in multiple other flow graphs
+     * @param vertex Vertex that should be cloned
+     * @param inputPin Input pin to the vertex from the previous vertices
+     * @param flow Flow between the input pin and the copied vertex
+     * @param previousNodeVertices List of previous vertices
+     * @return Returns a list of cloned vertices required for usage in multiple flow graphs
+     */
+    private List<DFDVertex> cloneFlowAndVertexForMultipleFlowGraphs(DFDVertex vertex, Pin inputPin, Flow flow, List<DFDVertex> previousNodeVertices) {
+        List<DFDVertex> newVertices = new ArrayList<>();
+
+        // If we run loop more then once, we may also need to copy flow, for the analysis
```

Comment:
Small typos, should be:
`If we run the loop more than once, we may also need to copy the flow, for the analysis`